File: gvpr/CSharp/GvprParserBase.cs
using Antlr4.Runtime;
using System.IO;

public abstract class GvprParserBase : Parser {
    private readonly ITokenStream _input;

    protected GvprParserBase(ITokenStream input, TextWriter output, TextWriter errorOutput)
        : base(input, output, errorOutput)
    {
        _input = input;
    }

    public bool IsSemiRequired()
    {
        var c = (this.InputStream as CommonTokenStream).LT(-1);
        var d = (this.InputStream as CommonTokenStream).LT(1);
        return c.Type != gvprParser.CCBC;
    }

    public bool IsSemiNotRequired()
    {
        var c = (this.InputStream as CommonTokenStream).LT(-1);
        var d = (this.InputStream as CommonTokenStream).LT(1);
        return c.Type == gvprParser.CCBC;
    }
}

File: gvpr/Dart/GvprParserBase.dart
import 'package:antlr4/antlr4.dart';
import 'dart:io';
import 'dart:convert';
import 'gvprParser.dart';

abstract class GvprParserBase extends Parser {
    GvprParserBase(TokenStream input) : super(input)
    {
    }

    bool IsSemiRequired()
    {
        var c = this.tokenStream.LT(-1);
        var d = this.tokenStream.LT(1);
        return c?.type != gvprParser.TOKEN_CCBC;
    }

    bool IsSemiNotRequired()
    {
        var c = this.tokenStream.LT(-1);
        var d = this.tokenStream.LT(1);
        return c?.type == gvprParser.TOKEN_CCBC;
    }
}
